src/builder.rs

```diff
@@ -924,7 +924,9 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
         // dereference after a drop, for instance.
         // FIXME(antoyo): this check that we don't call get_aligned() a second time on a type.
         // Ideally, we shouldn't need to do this check.
-        let aligned_type = if pointee_ty == self.cx.u128_type || pointee_ty == self.cx.i128_type {
+        let aligned_type = if (pointee_ty == self.cx.u128_type || pointee_ty == self.cx.i128_type)
+            && align == Align::from_bytes(16).unwrap()
```

Comment:
Instead of hard-coding the alignment, could you query it from the Rust compiler like we do [here](https://github.com/rust-lang/rustc_codegen_gcc/blob/9aec231fbaf09ecd9ee6350100045cfccd4f4689/src/context.rs#L156)?
Seems it is possibly executed many times, it might be better to store this result in `CodegenCx`.